.github/workflows/build.yml

```diff
@@ -23,35 +23,66 @@ jobs:
         with:
           node-version: ${{ matrix.node }}
           cache: yarn
-      - run: yarn
+      - run: yarn install --ignore-scripts
+      - run: yarn prebuild
+      - run: yarn build:cjs
       - run: tar -czf /tmp/web3-${{ matrix.node }}.js.tar.gz --exclude="./.git" ./
       - uses: actions/upload-artifact@v3
         with:
           name: web3-${{ matrix.node }}.js.tar.gz
           path: /tmp/web3-${{ matrix.node }}.js.tar.gz
-
+  build-cjs-types:
+      name: Build ESM & Types
+      needs: build
+      runs-on: ubuntu-latest
+      strategy:
+          matrix:
+              node: [ 18, 16 ]
+      steps:
+          -   uses: actions/setup-node@v3
+              with:
+                node-version: ${{ matrix.node }}
+          -   uses: actions/download-artifact@v3
+              with:
+                name: web3-${{ matrix.node }}.js.tar.gz
+                path: /tmp
+          -   run: tar -xf /tmp/web3-${{ matrix.node }}.js.tar.gz -C ./
+          -   run: yarn build:esm
+          -   run: yarn build:types
   lint:
-    name: lint
-    needs: build
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/download-artifact@v3
-        with:
-          name: web3-14.js.tar.gz
-          path: /tmp
-      - run: tar -xf /tmp/web3-14.js.tar.gz -C ./
-      - run: yarn lint
+      name: Lint
+      needs: build
+      runs-on: ubuntu-latest
+      strategy:
+          matrix:
+              node: [ 18 ]
```

Comment:
matrix should only be used when we want to run job for multiple variables, so it should be removed, as Lint is being run for one node version.

Reply:
done